src/Database/InfluxDB/JSON.hs

```diff
@@ -79,17 +80,17 @@ parseResultsWithDecoder Decoder {..} row val0 = success <|> parseErrorObject val
   where
     success = do
       results <- parseResultsObject val0
-      if results == V.fromList [A.emptyObject]
-        then return V.empty
-        else do
-          (join -> series) <- V.forM results $ \val ->
-            parseSeriesObject val <|> parseErrorObject val
-          values <- V.forM series $ \val -> do
-            (name, tags, columns, values) <- parseSeriesBody val
-            decodeFold $ V.forM values $ A.withArray "values" $ \fields -> do
-              assert (V.length columns == V.length fields) $ return ()
-              decodeEach $ row name tags columns fields
-          return $! join values
+
+      series <- fmap (join . V.map V.concat) $ V.forM results $ \val -> do
+        fmap pure (parseSeriesObject val)
+        <|> [] <$ parseStatementIdObject val
```

Comment:
This seems to be a bit complicated. I wonder if it's simpler if `parseSeriesObject` accepts empty objects.

Can you try something like this?

```diff
diff --git a/src/Database/InfluxDB/JSON.hs b/src/Database/InfluxDB/JSON.hs
index 4af9144..508dd44 100644
--- a/src/Database/InfluxDB/JSON.hs
+++ b/src/Database/InfluxDB/JSON.hs
@@ -81,10 +81,8 @@ parseResultsWithDecoder Decoder {..} row val0 = success <|> parseErrorObject val
     success = do
       results <- parseResultsObject val0

-      series <- fmap (join . V.map V.concat) $ V.forM results $ \val -> do
-        fmap pure (parseSeriesObject val)
-        <|> [] <$ parseStatementIdObject val
-        <|> parseErrorObject val
+      (join -> series) <- V.forM results $ \val ->
+        parseSeriesObject val <|> parseErrorObject val
       values <- V.forM series $ \val -> do
         (name, tags, columns, values) <- parseSeriesBody val
         decodeFold $ V.forM values $ A.withArray "values" $ \fields -> do
@@ -141,16 +139,12 @@ getTag tag tags = case HashMap.lookup tag tags of
   Just val -> return val

 parseResultsObject :: Value -> A.Parser (Vector A.Value)
-parseResultsObject = A.withObject "results" $ \obj -> obj .: "results"
+parseResultsObject = A.withObject "results" $ \obj -> do
+  fromMaybe V.empty <$> obj .:? "results"

 parseSeriesObject :: Value -> A.Parser (Vector A.Value)
 parseSeriesObject = A.withObject "series" $ \obj -> obj .: "series"

-parseStatementIdObject :: Value -> A.Parser Int
-parseStatementIdObject = withObject "results" $ \obj -> do
-  v <- obj .: "statement_id"
-  A.withScientific "statement_id" (return . round) v
-
 parseSeriesBody
   :: Value
   -> A.Parser (Maybe Text, HashMap Text Text, Vector Text, Array)
```